Libraries/LibWeb/XML/XMLDocumentBuilder.cpp

```diff
@@ -113,7 +113,18 @@ void XMLDocumentBuilder::element_start(const XML::Name& name, HashMap<XML::Name,
         m_namespace = Namespace::HTML;
     }
 
-    auto node = DOM::create_element(m_document, MUST(FlyString::from_deprecated_fly_string(name)), m_namespace).release_value_but_fixme_should_propagate_errors();
+    auto qualified_name = MUST(FlyString::from_deprecated_fly_string(name));
+
+    Optional<FlyString> prefix = {};
+    auto local_name = qualified_name;
+
+    if (qualified_name.bytes_as_string_view().contains(':')) {
+        auto parts = qualified_name.bytes_as_string_view().split_view(':');
+        prefix = MUST(FlyString::from_utf8(parts[0]));
+        local_name = MUST(FlyString::from_utf8(parts[1]));
+    }
```

Comment:
Would it be possible to use DOM::validate_and_extract() to do this instead? If an error is returned you can set `m_has_error = true` and return as above.